components/director/internal/domain/application/service.go

```diff
@@ -424,7 +435,18 @@ func (s *service) createRelatedResources(ctx context.Context, in model.Applicati
 
 	var apis []*model.APIDefinition
 	for _, item := range in.Apis {
-		apis = append(apis, item.ToAPIDefinition(s.uidService.Generate(), applicationID))
+
+		apiDefID := s.uidService.Generate()
+
+		var fetchRequestID *string
```

Comment:
You can extract this code to private method, but it's ok.

Reply:
It would be hard, as I already have private method `createFetchRequest` and there would be multiple inputs